File: src/pivotArray.c
// 2161. Partition Array According to Given Pivot
#include <stdlib.h>

int* pivotArray(int* nums, int numsSize, int pivot, int* returnSize) {
    int* result = (int*)malloc(numsSize * sizeof(int));
    *returnSize = numsSize;
    int idx = 0;

    for (int i = 0; i < numsSize; i++)
        if (nums[i] < pivot)
            result[idx++] = nums[i];

    for (int i = 0; i < numsSize; i++)
        if (nums[i] == pivot)
            result[idx++] = nums[i];

    for (int i = 0; i < numsSize; i++)
        if (nums[i] > pivot)
            result[idx++] = nums[i];

    return result;
}

File: src/pivotArray.py
class Solution:
    def pivotArray(self, nums: list[int], pivot: int) -> list[int]:
        lt = []
        gt = []
        eq = 0
        for num in nums:
            if num < pivot:
                lt.append(num)
            elif num > pivot:
                gt.append(num)
            else:
                eq += 1

        return lt + [pivot] * eq + gt
